packages/service-library/src/servicelib/logging_utils.py

```diff
@@ -169,67 +169,73 @@ def _log_arguments(
     return extra_args
 
 
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+@overload
+def log_decorator(
+    logger: logging.Logger | None = None, level: int = logging.DEBUG
+) -> Callable[[Callable[P, R]], Callable[P, R]]:
+    ...
+
+
+@overload
 def log_decorator(
-    logger=None, level: int = logging.DEBUG, *, log_traceback: bool = False
+    logger: logging.Logger | None = None, level: int = logging.DEBUG
+) -> Callable[
+    [Callable[P, Coroutine[Any, Any, R]]], Callable[P, Coroutine[Any, Any, R]]
+]:
+    ...
+
+
+def log_decorator(
```

Comment:
- I noticed that by only annotating thisone it was sufficient in my case
- THOUGHT In decorators perhaps we should use aliases since there is a lot of redundancy, specially with the Callables part

Reply:
@pcrespov if you are inerested, I think I managed something better